### PR TITLE
Bump Kubernetes dependencies to v1.29.3

### LINKED ISSUE #167: custom-metrics-apiserver compile failed with k8s.io/apiserver v0.29.2
HI, my go application using
```
sigs.k8s.io/custom-metrics-apiserver v1.28.0
k8s.io/apiserver v0.29.2
```
and it can't compile succeed because of this function **func (o *CustomMetricsAdapterServerOptions) ApplyTo(serverConfig *genericapiserver.Config) error **.  In this function, it called _ApplyTo_  in package _k8s.io/apiserver_,  but in v0.29.2version,  _ApplyTo_ has 3 parameters.  Also I compare  _k8s.io/apiserver_  between v0.29 and v0.28,  in version0.28, _ApplyTo_ has only 1 parameter.

Maybe I think custom-metrics-apiserver should fit kubernetes v1.29

### PR DESCRIPTION
**Pull request by dgrisonnet**
Fixes #167

### CODE REVIEW

**Comment by dgrisonnet** [comment]
Holding until I test it in prometheus-adapter.

/hold

File: go.mod [hunk 1]
```diff
@@ -5,18 +5,18 @@ go 1.21
 toolchain go1.21.5
```
Comment (RainbowMango) [inline go.mod:5]: Kubernetes v1.29.3 is built with go 1.21.8.
Shall we bump the toolchain to that version? I guess we'd better do it, but no big deal.

**Comment by CatherineF-dev** [comment]
It has a panic error inside pull-custom-metrics-apiserver-test 

```
--- FAIL: TestApplyTo (1.71s)
    --- FAIL: TestApplyTo/only-secure-port (1.71s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1577a9d]
goroutine 132 [running]:
testing.tRunner.func1.2({0x1d40020, 0x34a7ad0})
	/usr/local/go/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1548 +0x397
panic({0x1d40020?, 0x34a7ad0?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
k8s.io/client-go/kubernetes.NewForConfig(0xc000190500?)
	/home/prow/go/pkg/mod/k8s.io/client-go@v0.29.3/kubernetes/clientset.go:461 +0x1d
sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options.(*CustomMetricsAdapterServerOptions).ApplyTo(0xc0002f3dc0, 0xc0007a9bf0)
	/home/prow/go/src/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options/options.go:103 +0x1cd
sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options.TestApplyTo.func1(0xc000682b60)
	/home/prow/go/src/sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options/options_test.go:115 +0x1ce
testing.tRunner(0xc000682b60, 0xc0003dc410)
	/usr/local/go/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine [131](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_custom-metrics-apiserver/168/pull-custom-metrics-apiserver-test/1773066633665843200#1:build-log.txt%3A131)
	/usr/local/go/src/testing/testing.go:1648 +0x3ad
FAIL	sigs.k8s.io/custom-metrics-apiserver/pkg/cmd/options	1.752s
ok  	sigs.k8s.io/custom-metrics-apiserver/pkg/dynamicmapper	0.016s
ok  	sigs.k8s.io/custom-metrics-apiserver/pkg/provider	0.011s
FAIL
make: *** [Makefile:108: test] Error 1
```

File: go.mod (hunk 1 above)
Comment (zroubalik) [reply go.mod:5]: agree

**Comment by kentquirk** [comment]
Hi. I'm trying to build a custom metrics adapter and it doesn't currently build (#167), and I can't find any combination of dependencies or versions that will let me keep working. I'm hoping this issue will clear those things up. 

I started investigating the test error, and the function NewRecommendedConfig called in TestApplyTo is returning a *server.RecommendedConfig that contains a *ClientConfig which is nil. I don't know enough to know how to fix this problem, since I'm not sure if that's supposed to be an acceptable result.

If I can help move this along, please let me know how.

**Review by zroubalik — COMMENTED** [review]
I did a bump to 1.29.4 here, PTAL: https://github.com/kubernetes-sigs/custom-metrics-apiserver/pull/170

**Pull request closed without merging**